lib/parameter_substitution/expression.rb

```diff
@@ -37,6 +37,20 @@ def substitution_parameter_names
       @expression_list.map_compact(&:parameter_name)
     end
 
+    def method_names
+      @expression_list.map_compact do |expression|
+        name_from_method_calls(expression)
+      end.flatten
+    end
+
+    def name_from_method_calls(expression)
```

Comment:
I think this method could be named a little better to be more descriptive of what it's doing. Something like `methods_used_by_expression`